### ISSUE TITLE
`unnecessary_parenthesis` not detected around literals and simple identifiers

### ISSUE THREAD

**Issue by oprypin** [dart-lang/linter#4354]
These unnecessary parentheses should be dropped, but they are not reported:

```dart
void f() {
  final a1 = '' + ('');
  final a2 = (a1.length) * 2;
  final a3 = ((a1.isEmpty), 2);
  final a4 = (1, (2));
}
```

**Comment by srawlins** [comment]
Fixed by https://github.com/dart-lang/sdk/commit/a18645763d12bfd4a5e3cc19a191063a62051e3e
